spark/src/main/java/org/opensearch/sql/spark/asyncquery/model/AsyncQueryJobMetadata.java

```diff
@@ -8,37 +8,83 @@
 package org.opensearch.sql.spark.asyncquery.model;
 
 import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.sql.spark.execution.statement.StatementModel.QUERY_ID;
 
 import com.google.gson.Gson;
 import java.io.IOException;
-import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.EqualsAndHashCode;
-import org.opensearch.common.xcontent.XContentFactory;
-import org.opensearch.common.xcontent.XContentType;
-import org.opensearch.core.xcontent.DeprecationHandler;
-import org.opensearch.core.xcontent.NamedXContentRegistry;
+import lombok.SneakyThrows;
 import org.opensearch.core.xcontent.XContentBuilder;
 import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.index.seqno.SequenceNumbers;
+import org.opensearch.sql.spark.execution.statestore.StateModel;
 
 /** This class models all the metadata required for a job. */
 @Data
-@AllArgsConstructor
-@EqualsAndHashCode
-public class AsyncQueryJobMetadata {
-  private String applicationId;
-  private String jobId;
-  private boolean isDropIndexQuery;
-  private String resultIndex;
+@EqualsAndHashCode(callSuper = false)
+public class AsyncQueryJobMetadata extends StateModel {
+  public static final String TYPE_JOBMETA = "jobmeta";
+
+  private final AsyncQueryId queryId;
```

Comment:
explicit add AsyncQueryId, currently we use EMR-S jobId as queryId, but after adding InteractiveSession, it is not always true.

* For batch-query, dispatcher get jobId using AsyncQueryId.
* For interactive-query, AsyncQueryId is same as StatementId.